File: unity/BlackjackUnity/Assets/Score.cs
/*
  Blackjack
  Copyright 2015 Sam Saint-Pettersen
  Released under the MIT/X11 License.

  C#/Unity implementation
*/
using UnityEngine;
using System;

/// <summary>
/// Score implements a score or information box.
/// </summary>
public class Score {
    private bool debug;
    private int posX;
    private int posY;
    private string score;

    /// <summary>
    /// Constructor for score.
    /// </summary>
    /// <param name="debug">Enable debug messages?</param>
    /// <param name="posX">X position for score box.</param>
    /// <param name="posY">Y position for score box.</param>
    public Score(bool debug, int posX, int posY) {
        this.debug = debug;
        this.posX = posX;
        this.posY = posY;
        Debugger.Emit(this.debug, String.Format("Created score counter at {0},{1}", posX, posY));
    }

    /// <summary>
    /// Emit a score or similar message.
    /// </summary>
    /// <param name="score">Score or similar mesage to emit.</param>
    public void Emit(string score) {
        this.Clear();
        this.score = score;
    }

    /// <summary>
    /// Emit a score.
    /// </summary>
    /// <param name="score">Score to emit.</param>
    public void Emit(int score) {
        this.Emit(score.ToString());
    }

    /// <summary>
    /// Clear the score box.
    /// </summary>
    public void Clear() {
        this.score = "";
    }

    /// <summary>
    /// Draw the score box.
    /// </summary>
    public void Draw() {
        Rect rect = new Rect(this.posX, this.posY, 400, 200);
        GUI.Label(rect, this.score);
    }
}

File: html5/enchant.js_ts/score.ts
/*
	Blackjack
	Copyright 2015 Sam Saint-Pettersen
	Released under the MIT/X11 License.
	
	HTML5/enchant.js TypeScript implementation
*/

/// <reference path="enchant.d.ts" />
/// <reference path="debug.ts" />

class Score {
	private box: Label;

	/**
	 * Score implements a score or information box.
	 * @constructor
	 * @param {boolean} debug - Enable debug messages?
	 * @param {number} posX - X position for score box.
	 * @param {number} posY - Y position for score box.
	*/
	constructor(debug: boolean, posX: number, posY: number) {
		this.box = new Label('');
		this.box.color = 'rgb(255, 255, 255)';
		this.box.font = '10pt verdana, sans-serif';
		this.box.x = posX;
		this.box.y = posY;
		this.box.width = 500;

		Debug.emit(debug, "Created score counter at ${posX},${posY}");
	}

	/**
	 * Emit a score or similar message.
	 * @param {any} score - Score or similar message to emit.
	*/
	public emit(score: any): void {
		this.clear();
		this.box.text = score.toString();
	}

	/**
	 * Clear the score box.
	*/
	public clear(): void {
		this.box.text = '';
	}

	/**
	 * Draw the score box.
	 * @returns {Label} The score box.
	*/
	public draw(): Label {
		return this.box;
	}
}
